rest_auth_toolkit/serializers.py

```diff
@@ -51,6 +54,31 @@ def create(self, validated_data):
         )
 
 
+class EmailConfirmationDeserializer(serializers.Serializer):
+    email = serializers.EmailField()
+    token = serializers.CharField()
+
+    def validate(self, data):
+        msg = None
+
+        try:
+            confirmation = EmailConfirmation.objects.get(
+                external_id=data['token'],
+                user__email=data['email'],
+            )
+            confirmation.confirm()
+        except EmailConfirmation.DoesNotExist:
+            msg = _('Invalid link')
+        except EmailConfirmation.IsExpired:
+            # FIXME it's not possible to register with the same email
```

Comment:
address this

Reply:
We should check for expired confirmation, but it shouldn't be an error related to a duplicate email, tho? A user shouldn't be able to create an account with an email related to another email, tho ?